### PR TITLE
feat(useBluetooth): new function

### LINKED ISSUE #1286: proposal: useBluetooth()
Although this API is not well supported, I feel like providing some sort of support for it out of the box for "future generations" would be really, really beneficial for bridging the gap between PWAs and native app experiences ...

https://developer.mozilla.org/en-US/docs/Web/API/Web_Bluetooth_API

Thoughts are very much welcome!

### PR DESCRIPTION
**Pull request by michealroberts**
### Description

A reactive for working with the [Web Bluetooth API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Bluetooth_API) which provides the ability to connect and interact with Bluetooth Low Energy peripherals.

The Web Bluetooth API lets websites discover and communicate with devices over the Bluetooth 4 wireless standard using the Generic Attribute Profile (GATT).

### Considerations

N.B. It is currently partially implemented in Android M, Chrome OS, Mac, and Windows 10. For a full overview of browser compatibility please see [Web Bluetooth API Browser Compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Web_Bluetooth_API#browser_compatibility)

N.B. There are a number of caveats to be aware of with the web bluetooth API specification. Please refer to the [Web Bluetooth W3C Draft Report](https://webbluetoothcg.github.io/web-bluetooth/) for numerous caveats around device detection and connection.

N.B. This API is not available in Web Workers (not exposed via WorkerNavigator).

Fixes #1286 

---

- [ ] Bug fix
- [X] New Feature
- [x] Documentation update
- [ ] Other

### CODE REVIEW

**Comment by michael-observerly** [comment]
Special mention to @JamieCurnow who helped with documentation and usage suggestions. Please if possible could we add him as a contributor 🙏

File: packages/core/useBluetooth/index.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      }
+      catch (err) {
+        error.value = err
+      }
+    }
+  }
+
+  tryOnMounted(() => {
+    if (device.value)
+      device.value.gatt?.disconnect()
```
Comment (antfu) [inline packages/core/useBluetooth/index.ts:113]: Disconnect on mounted? Is this intentional?

Comment (michael-observerly) [reply packages/core/useBluetooth/index.ts:113]: Ahhh this should be connect()

**Comment by michael-observerly** [comment]
@antfu Any thought on this one? Close, or does this need further review?

**Comment by michealroberts** [comment]
@antfu Awesome. Rebased onto the latest main branch ... let me know if you need anything else! 👍

**Comment by michealroberts** [comment]
@antfu Linting and Testing workflows seem to be stuck ...

**Comment by antfu** [comment]
image

It still marks conflicting. You may want to merge with main instead of force push.

**Comment by michealroberts** [comment]
@antfu Let me know how it looks now ... fixed a few of the remaining issues from workflows. 👍

**Comment by michealroberts** [comment]
@okxiaoliang4 I think this one is ready now to go ... thoughts?

**Comment by antfu** [comment]
It still seems unmergeable to me. Maybe you could create a fresh new branch from the current main and open another PR? Thanks.

**Comment by michealroberts** [comment]
@antfu closing this - new PR is here #1694

**Pull request closed without merging**